### PR TITLE
Change to how event callbacks are dispatched and how logs are filtered

### PR DESCRIPTION
**Pull request by epheph**
Log filtering is now done with a single filter object, instead of one per contract
Based on this, refactored how we look up the results and the map required to perform this lookup
Changes the arguments and required data structures for startBlockchainEventListeners, should be major version bump

### CODE REVIEW

**Comment by ryanberckmans** [comment]
Travis:
image

File: src/events/start-blockchain-event-listeners.js [hunk 1]
```diff
@@ -1,41 +1,83 @@
...
+          eventName: eventName,
+          abi: eventAbi.inputs,
+        };
+        topics.add(eventAbi.signature);
+      }
+    }
+  }
+  const filter = {
+    address: addresses,
+    topics: [Array.from(topics)],
```
Comment (ryanberckmans) [inline src/events/start-blockchain-event-listeners.js]: Double checked what I'm sure you already knew: augur.js supports `Array.from` via build using babel/polyfill.

File: src/events/start-blockchain-event-listeners.js [hunk 2]
```diff
@@ -1,41 +1,83 @@
...
+  return compact(parsedLogs);
+}
+
+// Iterates through eventCallbacks, generates eth filter and event signature mapping
+function getEventsDetails(eventsToSubscribe, activeContracts) {
+  var eventByAddressAndSignature = {};
+  const addresses = [];
+  const topics = new Set();
+  for (const contractName in eventsToSubscribe) {
+    if (Object.prototype.hasOwnProperty.call(eventsToSubscribe, contractName)) {
```
Comment (ryanberckmans) [inline src/events/start-blockchain-event-listeners.js]: Perhaps `if (!Object.prototype.hasOwnProperty.call(eventsToSubscribe, contractName)) { continue; }` to lower cyclomatic complexity of inner loop

Comment (epheph) [reply src/events/start-blockchain-event-listeners.js]: We use eslint rule `no-continue`

**Pull request merged**